src/config/api/api.types.ts

```diff
@@ -0,0 +1,24 @@
+export interface RequestParams<T> {
+  searchText?: string;
+  page?: number;
+  pageSize?: number;
+  sort?: string;
+  filter?: T;
+  with?: string | string[];
+}
+
+export interface ServicePagination {
+  count: number;
+  currentPage: number;
+  links: { next: string; previous: string };
+  perPage: number;
+  total: number;
+  totalPages: number;
+}
+
+export interface ServiceResponse<T> {
+  data: T;
+  pagination: ServicePagination;
```

Comment:
If we are planning to use `ServiceResponse` also for error responses maybe `pagination` should be optional? Even for the success case, I can see scenarios where we don't want to paginate the results.

In case we also use these interface for errors we should talk about what would the `T` type looks like in that case.

Reply:
The success and error responses are slightly different, so it would make sense to have separate interfaces.

Successful response:
```
{
    "status": 200,
    "success": true,
    "data": null
}
```

Error response:
```
{
    "success": false,
    "status": 500,
    "error": {
        "code": null,
        "message": null
    }
}
```

Also, `pagination` is only present in successful responses.

Reply:
In my opinion, it would be best to have three interfaces. One for success responses without pagination, one for success responses with pagination, and one for error responses.
```
export interface SuccesfulServiceResponse<T> {
  status: number;
  success: true;
  data: T;
}

export interface PaginatedServiceResponse<T> extends ServiceResponse<T> {
  pagination: PaginationData;
}

export interface ErrorServiceResponse<T> {
  status: number;
  success: false;
  error: {
    code: string|null;
    message: string|null;
  };
}
```

Reply:
Agreed!

Reply:
> It would be great for ServiceResponse to be an union of both success and error response

+1